### PR TITLE
Dispatch Plot improvements

### LINKED ISSUE #322: HERON Visualization Updates
--------
Issue Description
1.  Dispatch plot visualization 
2.  Capacity plot visualization 
--------
##### Please attach the input file(s) that generate this error.  The simpler the input, the faster we can find the issue.

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [x] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by PaulTalbot-INL**
--------
Pull Request Description
--------
##### What issue does this change request address?
Closes #322

##### What are the significant changes in functionality due to this change request?
Several updates to DispatchPlot:
- Colors for a single component are the same across resource plots.
- Color schemes updated to increase clarity and distinguishability within each resource plot
- If the data has 3 resources, it keeps it all on one plot; otherwise, it puts 2 axes per figure and makes more figures
- Adds hatching to extend the number of components that can be represented. Hatching is only used once the first set of colors are exhausted, at which point it loops through the colors again with each new hatch. This allows for (10x9)=90 components to be represented. If over 90 components are present, a warning is raised.
- Removes `title()` modification on component names (but keeps `"_"` to `" "` swapping)

See examples below.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/HERON/wiki/Code-Standards) for details.
- [x] 4. Automated Tests should pass.
- [x] 5. If significant functionality is added, there must be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large tes.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added, the the analytic documentation must be updated/added.
- [x] 9. If any test used as a basis for documentation examples have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

**Comment by PaulTalbot-INL** [comment]
For reference, here are a few examples of new plots:
dispatch_id0_y2020_c0_f1
dispatch_id0_y2020_c0_f2

File: src/DispatchPlot.py [hunk 1]
```diff
@@ -219,48 +255,41 @@ def plot_signal(self, fig, axes, df, sid, mstep, cid) -> None:
   def color_style(self, grp_vars):
     """
       @ In, grp_vars, Dict[str, List[str]], a dictionary mapping components to variables.
```
Comment (dylanjm) [inline src/DispatchPlot.py:258]: I think there is a missing description for this doc string.

File: tests/integration_tests/mechanics/debug_mode/tests [hunk 2]
```diff
@@ -15,7 +15,7 @@
     [../]
     [./debug_plot]
       type = Exists
-      output = 'Debug_Run_o/dispatch_id0_y10_c0.png Debug_Run_o/dispatch_id0_y11_c0.png Debug_Run_o/dispatch_id1_y10_c0.png Debug_Run_o/dispatch_id1_y11_c0.png'
+      output = 'Debug_Run_o/dispatch_id0_y10_c0_f1.png Debug_Run_o/dispatch_id0_y11_c0_f1.png Debug_Run_o/dispatch_id1_y10_c0_f1.png Debug_Run_o/dispatch_id1_y11_c0_f1.png'
```
Comment (dylanjm) [inline tests/integration_tests/mechanics/debug_mode/tests:18]: I didn't catch what the `f1` stands for now. Is that when you have >3 resources it creates a new figure?

**Review by dylanjm — CHANGES_REQUESTED** [review]
Very minor change that I believe was missed in a previous PR.

File: tests/integration_tests/mechanics/debug_mode/tests (hunk 2 above)
Comment (PaulTalbot-INL) [reply tests/integration_tests/mechanics/debug_mode/tests:18]: yeah, open to other ideas, in my head it's "figure M/N" but I decided against including "N".

**Review by dylanjm — APPROVED** [review]
Changes are good. Tests were previously passing and the only change in the latest commit modified just documentation. Approved to merge.

**Pull request merged**